File: bun-resend-examples/typescript/examples/with-template.ts
import { Resend } from "resend";

const resend = new Resend(process.env.RESEND_API_KEY);

const from = process.env.EMAIL_FROM || "Acme <onboarding@resend.dev>";
const templateId = process.env.RESEND_TEMPLATE_ID || "your-template-id";

// Send email using a Resend hosted template
// Template variables must match exactly (case-sensitive)
// Do not use html or text when using a template
const { data, error } = await resend.emails.send({
  from,
  to: ["delivered@resend.dev"],
  subject: "Email from Template",
});

if (error) {
  console.error("Error sending email:", error);
  process.exit(1);
}

console.log("Email sent successfully!");
console.log("Email ID:", data?.id);
console.log("Template ID:", templateId);

File: rust-resend-examples/examples/with_template.rs
use resend_rs::types::CreateEmailBaseOptions;
use resend_rs::Resend;

#[tokio::main]
async fn main() {
    dotenvy::dotenv().ok();

    let api_key = std::env::var("RESEND_API_KEY").expect("RESEND_API_KEY environment variable is required");
    let from = std::env::var("EMAIL_FROM").unwrap_or_else(|_| "Acme <onboarding@resend.dev>".to_string());
    let template_id = std::env::var("RESEND_TEMPLATE_ID").unwrap_or_else(|_| "your-template-id".to_string());

    let resend = Resend::new(&api_key);

    // Send email using a Resend hosted template
    // Template variables must match exactly (case-sensitive)
    // Do not use html or text when using a template
    // Note: Check the latest Rust SDK docs for template support syntax
    let email = CreateEmailBaseOptions::new(&from, ["delivered@resend.dev"], "Email from Template - Rust Example");

    match resend.emails.send(email).await {
        Ok(response) => {
            println!("Email sent successfully!");
            println!("Email ID: {}", response.id);
            println!("Template ID: {}", template_id);
        }
        Err(e) => {
            eprintln!("Error sending email: {}", e);
            std::process::exit(1);
        }
    }
}
